ros2pkg/ros2pkg/api/create_api.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import em
+from io import StringIO
+import os
+import sys
+
+
+def expand_template(template_file, data, output_file, minimum_timestamp=None):
+    output = StringIO()
+    interpreter = em.Interpreter(
+        output=output,
+        options={
+            em.BUFFERED_OPT: True,
+            em.RAW_OPT: True,
+        },
+        globals=data,
+    )
+    with open(template_file, 'r') as h:
+        try:
+            interpreter.file(h)
+        except Exception:
+            if os.path.exists(output_file):
+                os.remove(output_file)
+            print("Exception when expanding '%s' into '%s'" %
+                  (template_file, output_file), file=sys.stderr)
+            raise
+    content = output.getvalue()
+    interpreter.shutdown()
+
+    # only overwrite file if necessary
+    # which is either when the timestamp is too old or when the content is different
+    if os.path.exists(output_file):
+        timestamp = os.path.getmtime(output_file)
+        if minimum_timestamp is None or timestamp > minimum_timestamp:
+            with open(output_file, 'r') as h:
+                if h.read() == content:
+                    return
+    else:
+        # create folder if necessary
+        try:
+            os.makedirs(os.path.dirname(output_file))
```

Comment:
`os.makedirs` has an optional argument `exist_ok` which can be used to avoid the try/except.